### PR TITLE
Add Binary Search and Linear Search

### PR DESCRIPTION
**Pull request by iamneha**

### CODE REVIEW

File: docs/source/searching.rst [hunk 1]
```diff
@@ -0,0 +1,16 @@
+.. _searching:
+
+Searching
+=========
+
+Searching is a process of finding items in already sorted list. This process can be implemented via many different algorithms. Following is the list of searching algorithm which are implemented in Pydsa
```
Comment (aktech) [inline docs/source/searching.rst]: * Searching doesn't require the list to be sorted.
* Line length is > than 79 chars
* Pydsa --> PyDSA

File: pydsa/binary_search.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+    first = 0
+    last = len(List) - 1
+
+    while first <= last:
+        mid = (first + last)//2
+        if item == List[mid]:
+            return mid
+        elif item < List[mid]:
+                last = mid
```
Comment (aktech) [inline pydsa/binary_search.py]: `mid` --> `mid - 1`

File: pydsa/linear_search.py [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+    >>> List = [3, 4, 6, 8, 12, 15, 26]
+    >>> item = 6
+    >>> linear_search(List, item)
+    2
+    """
+    
+    index = len(List)
+    for i in range(0, index):
+        if List[i] == item:
+            return i
```
Comment (aktech) [inline pydsa/linear_search.py]: It should return -1 when the element is not found.

File: pydsa/binary_search.py [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+        else:
+                first = mid + 1
+    return None
```
Comment (aktech) [inline pydsa/binary_search.py]: It should return -1 when the element is not found.

File: pydsa/tests/test_binary_search.py [hunk 5]
```diff
@@ -0,0 +1,5 @@
+from pydsa.binary_search import binary_search
+def test_binary_search():
```
Comment (aktech) [inline pydsa/tests/test_binary_search.py]: Leave a couple of lines after import. Follow PEP8 standards.

File: pydsa/tests/test_linear_search.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from pydsa.linear_search import linear_search
+def test_linear_search():
```
Comment (aktech) [inline pydsa/tests/test_linear_search.py]: Same here https://github.com/pydsa/pydsa/pull/37/files#r57843217

File: pydsa/tests/test_linear_search.py [hunk 7]
```diff
@@ -0,0 +1,5 @@
...
+    List = [3, 6, 9, 11, 14, 16]
```
Comment (aktech) [inline pydsa/tests/test_linear_search.py]: use a random list, see other tests for motivation.

File: pydsa/tests/test_binary_search.py [hunk 8]
```diff
@@ -0,0 +1,5 @@
...
+        List = [1, 3, 5, 7, 9, 10]
+        item = 5
+        assert(binary_search(List, item) == 2)
```
Comment (aktech) [inline pydsa/tests/test_binary_search.py]: No need to use parenthesis for assertion.

**Comment by aktech** [comment]
@iamneha  Thanks for the PR, I have made some comments to improve this.

**Comment by iamneha** [comment]
I improve this, for further changes let me know asap.

**Pull request closed without merging**